src/Microsoft.DotNet.Build.CloudTestTasks/AzureHelper.cs

```diff
@@ -290,7 +290,7 @@ private static bool IsWithinRetryRange(HttpStatusCode statusCode)
                 response.Dispose();
             }
 
-            throw new HttpRequestException(string.Format("Request failed with status {0} response {1}", statusCode, contentStr));
+            throw new HttpRequestException(string.Format("Request failed with status {0} response {1} request url {2}", statusCode, contentStr, createRequest().RequestUri));
```

Comment:
You could probably just move this to an interpolated string for readability - `$"Request failed with status {statusCode} response {contentStr} ..."`